sources/packages/api/src/services/application/application.service.ts

```diff
@@ -1164,4 +1164,43 @@ export class ApplicationService extends RecordDataModelService<Application> {
       select: ["id"],
     }));
   }
+
+  /**
+   * Retrieve application with application number in
+   * completed status.
+   ** Application id is used to perform same lookup by id
+   ** instead of application number.
+   * @param applicationNumber
+   * @param userId
+   * @param applicationId
+   * @returns
+   */
+  async getApplicationToRequestChange(
+    applicationNumber: string,
+    userId: number,
+    applicationId?: number,
+  ): Promise<Application> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select(["application.id", "application.applicationNumber"])
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where("user.id = :userId", { userId })
+      .andWhere("application.applicationStatus = :completed", {
+        completed: ApplicationStatus.completed,
+      });
+    if (applicationId) {
+      applicationQuery.andWhere("application.id = :applicationId", {
+        applicationId,
+      });
+    } else {
+      applicationQuery.andWhere(
+        "application.applicationNumber = :applicationNumber",
```

Comment:
It seems that the `applicationNumber` is optional sometimes, please check if it is defined before use it.